### ISSUE TITLE
[Bug] danawa-price-search: 결제조건 배지(.ico.cash 등) 미캡처로 현금 전용가가 일반 최저가로 오표시

### ISSUE THREAD

**Issue by Combi153** [NomaDamas/k-skill#252]
## 증상

 `danawa-price-search` 스킬의 `offers`로 `pcode=75001853` (갤럭시 S25 256GB 자급제) 가격비교를 호출하면 1위로 **킴스클럽 979,000원**이 잡힙니다. 그런데 다나와 페이지에서 같은 가격은 `<span class="ico cash">현금</span>` 배지가 붙은 **현금 결제 전용가**(카드 결제 불가)입니다. 카드 결제 기준 실제 최저가는 롯데ON 1,073,890원입니다.

 결과적으로 비교 표는 사용자에게 충분한 정보로 "최저가"를 안내하지 못합니다.

<img width="961" height="404" alt="Image" src="https://github.com/user-attachments/assets/0cc9e405-5260-41c5-8d0a-6b8dbd9dc5b8" />

<img width="965" height="671" alt="Image" src="https://github.com/user-attachments/assets/11eab229-7276-4f93-a598-a7c7e38cf59c" />

결제조건 배지 등을 캡처해서 결제 정보로 "현금 전용가" 등 정보를 더 준다면 좋겠습니다.

**Comment by vkehfdl1** [comment]
## AI-understood issue summary

`danawa-price-search offers 75001853` currently ranks 킴스클럽 `979,000원` as the lowest offer, but the live Danawa price-compare fragment marks that row with `<span class="ico cash">현금</span>` and the item text says `(개인소득공제용 현금영수증 100%발행)`. The helper output does not expose that payment-condition badge, so a cash-only/special-payment offer can be presented as the general lowest price instead of being clearly labeled or separated from card/general-commerce prices.

## Why this issue is needed

- The skill promises conservative comparison of delivery-inclusive prices, card discounts, and installment/payment-relevant terms in `danawa-price-search/SKILL.md:15-20` and documents card/payment-sensitive presentation in `danawa-price-search/SKILL.md:133-147`.
- The current parser already reads the same `.diff_item` rows from Danawa’s price AJAX surface, but only extracts mall, price, shipping, card line, installment, and URL; it ignores non-card badges such as `.ico.cash` in the price line (`danawa-price-search/scripts/danawa_search.py:189-224`).
- A live read-only check of `python3 danawa-price-search/scripts/danawa_search.py offers 75001853 --limit 5` reproduced the problem: 킴스클럽 `979,000원` is first, with no payment-condition fields in the JSON. A separate read-only fetch of the same AJAX fragment showed the 킴스클럽 row under `id="cashMallList"`, with `<span class="ico cash">현금</span>` and the cash-receipt-only product text.

## Why this issue may not be needed

- If the intended product contract is “show Danawa’s absolute lowest displayed offer, regardless of payment restrictions,” the existing sort is consistent with Danawa’s displayed minimum price.
- The issue is presentation/interpretation rather than a complete data outage: the offer still includes the mall, price, shipping, and buy link.
- Danawa’s HTML classes are not a formal public API, so adding more selectors increases maintenance surface when Danawa changes markup.

## Expected Outcome

For `offers` and `compare`, each offer should include Danawa-exposed payment-condition metadata such as cash-only badges. User-facing output should make cash-only/special-payment offers visible (e.g. “현금 전용가”) and avoid implying that such rows are ordinary card/general lowest prices. If card/general ranking is needed, the implementation should either compute it separately or document/filter behavior clearly.

## Evidence-based implementation plan

### Feasibility from repo research

Reusable code exists in-repo:

- `danawa-price-search/scripts/danawa_search.py:46-53` already centralizes BeautifulSoup creation and text cleanup, so badge extraction can reuse existing parser helpers.
- `danawa-price-search/scripts/danawa_search.py:155-228` is the single `offers()` path that builds the POST payload, parses `.diff_item`, and returns normalized offer dictionaries. This is the right insertion point for payment-condition fields.
- `danawa-price-search/scripts/danawa_search.py:199-222` already extracts nearby payment-relevant details (`.card_line`, `.btn_foi`, `.foi_layer`), so adding `.prc_line .ico` / `.ico.cash` extraction should fit the current shape without a new package.
- `danawa-price-search/SKILL.md:101-116` and `docs/features/danawa-price-search.md:27-41` document the current offer fields; both should be updated if new JSON fields are added.
- `danawa-price-search/SKILL.md:159-166` and `docs/features/danawa-price-search.md:43-47` already document selector drift as a known failure mode.
- `docs/roadmap.md:142-145` says a stable public Danawa CLI had not been found earlier, supporting a targeted parser fix rather than replacing the skill.

External research:

- Danawa live price-compare AJAX fragment - https://prod.danawa.com/info/ajax/getAllPriceCompareMallList.ajax.php : the endpoint returns HTML fragments containing `.diff_item` rows; for `pcode=75001853`, the problematic row was observed with `id="cashMallList"`, `.ico.cash`, and cash-receipt-only product text.
- Beautiful Soup Documentation - https://www.crummy.com/software/BeautifulSoup/bs4/doc/ : Beautiful Soup supports CSS selectors and `select_one()`, matching the parser’s current approach.
- beautifulsoup4 · PyPI - https://pypi.org/project/beautifulsoup4/ : the existing dependency remains suitable for this targeted HTML-fragment parsing task.
- no suitable external library found for a maintained Danawa-specific price/offers parser that already exposes these cash/payment badges; the implementation should extend the existing helper instead of adding a new dependency.

### Step-by-step instructions for the implementation agent

1. Add a small payment-condition extraction helper in `danawa-price-search/scripts/danawa_search.py`, near `clean_text()` / `parse_int()`:
   - Input: a `.diff_item` BeautifulSoup tag.
   - Extract badges from the price line, likely `div.select(".prc_line .ico")`, preserving text and class names.
   - Treat a badge with class `cash` or text containing `현금` as cash-only/special cash payment.
   - Also consider capturing the `.info_line .link` text because the live row’s product text contains `(개인소득공제용 현금영수증 100%발행)`.

2. Extend each offer dictionary in `offers()` (`danawa-price-search/scripts/danawa_search.py:206-224`) with explicit fields. Suggested names:
   - `payment_badges`: list of badge labels, e.g. `["현금"]`.
   - `payment_badge_classes`: optional list/list-of-lists for raw class evidence, e.g. `["ico cash"]` or `[ ["ico", "cash"] ]`.
   - `payment_condition`: normalized summary, e.g. `"현금 전용가"` when `.ico.cash` or `현금` is present.
   - `is_cash_only`: boolean, true for `.ico.cash` / `현금` badge.
   - `offer_title` or `product_line`: optional normalized `.info_line .link` text, useful for surfacing cash-receipt-only restrictions.

3. Keep the default sorting conservative:
   - Do not silently drop cash-only rows unless a new option is explicitly added.
   - If adding a separate “card/general lowest” calculation, keep it as additional metadata, not a breaking replacement for current `total_price` sorting.
   - Risk: `cashMallList` may not mean every row is card-impossible in all Danawa categories; prefer row-level `.ico.cash`/text evidence over only ancestor IDs.

4. Add regression coverage around the raw HTML shape:
   - There is no dedicated Danawa test file found today; create a small fixture containing one normal `.diff_item` and one `.ico cash` `.diff_item`, and test the parser/helper if the project’s Python test layout supports it.
   - Avoid tests that depend on live Danawa network responses; use fixture HTML so CI is deterministic.
   - Do not add tests asserting `.changeset/*.md` existence or fixed package versions (repo anti-patterns).

5. Update docs in the same change:
   - `danawa-price-search/SKILL.md:101-116` should list the new payment-condition fields.
   - `danawa-price-search/SKILL.md:122-147` should update the example table/summary to include a “결제조건” column or footnote such as “현금 전용가”.
   - `docs/features/danawa-price-search.md:27-41` should document the same fields and interpretation.

6. Verification for the implementation agent:
   - Run a fixture/unit test for the `.ico cash` parsing path.
   - Run `python3 danawa-price-search/scripts/danawa_search.py offers 75001853 --limit 5` manually as a smoke check if network is available; verify 킴스클럽 includes `is_cash_only: true` / `payment_condition: "현금 전용가"` or equivalent.
   - For release/docs/package-adjacent changes, run `npm run ci` per repo instructions if feasible; if too slow or environment-blocked, record the gap in the PR.

### Risks / assumptions from research

- Danawa markup is not a stable public API; selector drift remains the main risk.
- The implementation should not automate buying, login, CAPTCHA bypass, or checkout behavior; the skill is read-only (`danawa-price-search/SKILL.md:29-35`).
- The exact output-field names are not yet human-approved; the implementation agent may choose the names above or compatible alternatives, but should keep the meaning self-evident and documented.

## Assumptions / human decisions to resolve asynchronously before `/approve`

- Assumption: cash-only/special-payment rows should remain visible but clearly labeled, not hidden by default.
- Human decision if desired: whether to add an explicit CLI option such as `--exclude-cash-only` or only expose metadata and let callers decide.

## Approval reminder

Implementation starts only after a human comment containing `/approve`. The implementation agent will use this comment plus the issue body as its handoff.

## Agent Signature
